sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -225,6 +234,62 @@ public int getSize(Message amqpMessage) {
         return amqpMessage;
     }
 
+    private Map<String, Object> convertToDescribedType(Map<String, Object> propertiesValue) {
+        for (Map.Entry<String, Object> entry : propertiesValue.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof URI) {
+                propertiesValue.put(entry.getKey(), new ServiceBusDescribedType(ServiceBusConstants.URI_SYMBOL, ((URI) value).toString()));
```

Comment:
Is it possible to have classes that are specific to the type they implement? (ie. UriDescribedType which takes a Uri). I think this moves all the weird conversion logic into that class itself, so we can test that class separately and not have these helper methods in this class